libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -1084,16 +1084,94 @@ def configurable_alternatives(
 
 
 class RunnableSequence(RunnableSerializable[Input, Output]):
-    """
-    A sequence of runnables, where the output of each is the input of the next.
+    """A sequence of runnables, where the output of each is the input of the next.
+
+    RunnableSequence is the most important composition operator in LangChain as it is
+    used in virtually every chain.
+
+    A RunnableSequence can be instantiated directly or more commonly by using the `|`
+    operator where either the left or right operands (or both) must be a Runnable.
+
+    Any RunnableSequence automatically supports sync, async, batch.
+
+    The default implementations of `batch` and `abatch` utilize threadpools and
+    asyncio gather and will be faster than naive invocation of invoke or ainvoke
+    for IO bound runnables.
+
+    Batching is implemented by invoking the batch method on each component of the
+    RunnableSequence in order.
+
+    A RunnableSequence preserves the streaming properties of its components, so if all
+    components of the sequence implement a `transform` method -- which
+    is the method that implements the logic to map a streaming input to a streaming
+    output -- then the sequence will be able to stream input to output!
+
+    If any component of the sequence does not implement `transform` then the sequence
+    will have to wait for all the components input to be generated, then wait
+    for the component generate output, before it can stream this component's output.
+
+    Please note: RunnableLambdas do not support `transform` by default! So if
```

Comment:
worth adding note?

if you want to implement custom functionality and support streaming, you'll need to create a new class that inherits from Runnable and specifies transform logic.

Reply:
yeah